README.md

```diff
@@ -0,0 +1,55 @@
+# Evergreen updater [![npm](https://img.shields.io/npm/v/@blinkmobile/evergreen.svg?maxAge=2592000)](https://www.npmjs.com/package/@blinkmobile/evergreen)
+
+A Module to handle getting and using an evergreen update made with the [Blink Buildbots](https://github.com/blinkmobile/buildbot-cli)
+
+## Installation
+
+`npm i --save @blinkmobile/evergreen`
+
+## Prerequisites
+
+- Apache Cordova
+- [ionic-plugin-deploy](https://www.npmjs.com/package/ionic-plugin-deploy)
+
+## Usage
+
+For iOS < 10 and older Android projects `dist/bm-evergreen-updater-legacy.js` should be used. This includes the [WhatWG fetch pollyfil](https://github.com/github/fetch) and is transpiled from ES6 to ES5. ES6 builds using `import` will include `src/evergreen-updater.js`
+
+You must make sure the deviceready event from cordova has fired before using this plugin
+
+```javascript
+const EvergreenUpdater = require('@blinkmobile/evergreen')
+
+document.addEventListener('deviceready', function () {
+  const eu = new EvergreenUpdater(window.IonicDeploy, 'appId', 'blinkTennantId', window.cordova.platformId.toLowerCase())
```

Comment:
Do you think this constructor might be better taking in a single parameter that is object with the defined properties: 
- `IonicDeploy`
- `appId`
- `tenantId`
-  `platformId`

This way it might be easier to default some of the arguments e.g. I assume we could get `IonicDeploy` and `platformId` from window internally if not specified in the constructor options?

Reply:
we need to make sure that the `deviceready` event has been run because until that event is fired, there are no guarantees that `window.IonicDeploy` or `window.cordova` exist. I dont want to do that detection in the constructor because that leads to a number of complications - i believe a constructor should set values, init the class and nothing else

so by requiring them to be passed in, we can be sure that the `deviceready` event has fired because the developer cant get at them before then

im not fussed on if its an object or just params - i find params to be better for constructors but thats personal preference i guess

Reply:
Shouldn't our constructor check for `window.IonicDeploy` and `window.cordova` and throw errors if these are not available?